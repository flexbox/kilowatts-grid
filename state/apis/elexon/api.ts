import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import * as pn from "./pn";
import * as mels from "./mels";
import * as boalf from "./boalf";

export const BASE_API = `https://data.elexon.co.uk/bmrs/api/v1`;

export const elexonApi = createApi({
  reducerPath: "elexonApi",
  baseQuery: fetchBaseQuery({ baseUrl: BASE_API }),
  endpoints: (builder) => ({
    pn: builder.query({
      query: pn.query,
      transformResponse: pn.transformResponse
    }),
    mels: builder.query({
      query: mels.query,
      transformResponse: mels.transformResponse
    }),
    boalf: builder.query({
      query: boalf.query,
      transformResponse: boalf.transformResponse
    }),
  }),
});


export const { usePnQuery, useMelsQuery, useLazyMelsQuery, useBoalfQuery } = elexonApi;