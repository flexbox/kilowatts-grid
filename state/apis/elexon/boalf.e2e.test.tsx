import { createStore } from "../../reducer";
import { elexonApi } from "./api";
import { Provider } from "react-redux";
import React from "react";
import { renderHook, cleanup, waitFor } from "@testing-library/react-native";

const { useBoalfQuery } = elexonApi;

const wrapper = (props: any) => {
  return <Provider store={createStore()} children={props.children} />;
};


describe("state/apis/elexon/boalf.e2e.test.ts", () => {
  test("useBoalfQuery", async() => {
    const p = {
      from: '2023-12-01T00:00:00.000Z',
      to: '2023-12-01T00:00:29.000Z',
    };
    const r = renderHook(
      () =>
        useBoalfQuery(p),
      { wrapper }
    );

    await waitFor(() => {
      expect(r.result.current.data).toBeDefined();
      expect(r.result.current.data?.length).toBeGreaterThan(0);
      expect(r.result.current.data).toMatchSnapshot();
    });

    r.unmount();
  }, 5000);

  afterEach(async () => {
    await cleanup();
  });
});
