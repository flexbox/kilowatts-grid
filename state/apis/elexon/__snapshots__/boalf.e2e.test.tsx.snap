// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state/apis/elexon/boalf.e2e.test.ts useBoalfQuery 1`] = `
[
  {
    "bmUnit": "T_MEDP-1",
    "boalfs": [
      {
        "acceptanceNumber": 126173,
        "acceptanceTime": "2023-11-30T23:58:00Z",
        "amendmentFlag": "ORI",
        "deemedBoFlag": false,
        "levels": [
          {
            "level": 706,
            "time": "2023-12-01T00:00:00Z",
          },
          {
            "level": 650,
            "time": "2023-12-01T00:03:00Z",
          },
        ],
        "rrFlag": false,
        "soFlag": false,
        "storFlag": false,
      },
    ],
  },
]
`;
